### PR TITLE
Add typescript to library

### PR DESCRIPTION
**Pull request by danielwinkler**

### CODE REVIEW

**Comment by gitstud** [comment]
Let's get these conflicts resolved and merged 💃 :)

**Comment by danielwinkler** [comment]
@gitstud this one is fixed  too!

File: src/WalkthroughElement.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import React, {FunctionComponent, ReactNode, ReactElement} from 'react';
+import Tooltip, {TooltipChildrenContext, TooltipProps} from 'react-native-walkthrough-tooltip';
+
+import {WalkthroughContext} from './ContextWrapper';
+
+type UseTooltipChildContextT =
```
Comment (jasongaare) [inline src/WalkthroughElement.tsx:6]: this type should have a lower case `t` at the end, right?

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "printWidth": 120,
+  "tabWidth": 2,
+  "useTabs": false,
+  "semi": true,
+  "singleQuote": true,
+  "trailingComma": "es5",
+  "arrowParens": "avoid",
+  "bracketSpacing": false
```
Comment (jasongaare) [inline .prettierrc:9]: personal pref, but could you set "bracketSpacing" to `true` please?

**Comment by gitstud** [comment]
@jasongaare This should help for publishing https://medium.com/cameron-nokes/the-30-second-guide-to-publishing-a-typescript-package-to-npm-89d93ff7bccd

File: .prettierrc (hunk 2 above)
Comment (danielwinkler) [reply .prettierrc]: also my preference, i changed it and reformatted the code :-)

File: src/WalkthroughElement.tsx (hunk 1 above)
Comment (danielwinkler) [reply src/WalkthroughElement.tsx]: the T should have indicated that it's a type, but i think without the T it's also fine as there is no name conflict

**Comment by danielwinkler** [comment]
@jasongaare I implemented your suggestions

i also added tsdx which should make the whole process quite easy, 
https://github.com/jaredpalmer/tsdx

I think it'll be sufficient to read the [Quick start](https://github.com/jaredpalmer/tsdx)

just build your TS project and it will create everything you need in the dist folder:
`yarn build
yarn run v1.21.1
$ tsdx build --format cjs,esm,umd --name react-native-walkthrough
✓ Creating entry file 2.8 secs
⠸ Building modules  54%                 1.8s, estimated 3.4s
No name was provided for external module 'react-native-walkthrough-tooltip' in output.globals – guessing 'Tooltip'
No name was provided for external module 'prop-types' in output.globals – guessing 'PropTypes'
No name was provided for external module 'events' in output.globals – guessing 'events'
Creating a browser bundle that depends on Node.js built-in module ('events'). You might need to include https://www.npmjs.com/package/rollup-plugin-node-builtins
No name was provided for external module 'react-native-walkthrough-tooltip' in output.globals – guessing 'Tooltip'
No name was provided for external module 'prop-types' in output.globals – guessing 'PropTypes'
No name was provided for external module 'events' in output.globals – guessing 'events'
✓ Building modules 2.6 secs
✨  Done in 6.54s.`

Then you can publish to npm (it would trigger the build above automatically if you forgot):
`yarn publish
yarn publish v1.21.1
[1/4] Bumping version...
info Current version: 0.1.0
question New version: 0.1.1
info New version: 0.1.1
[2/4] Logging in...`

**Comment by danielwinkler** [comment]
@gitstud Thanks for your support !

**Comment by jasongaare** [comment]
I apologize it's been hard for me to find time for this project lately...

Finally released as version `0.2.0` 🎉

**Comment by danielwinkler** [comment]
@jasongaare 

no worries, I really appreciate your work and totally understand that it's a hard task to maintain a side project!

Cheers,
Daniel

**Pull request merged**